### PR TITLE
Add a libsys status report

### PR DESCRIPTION
**Pull request by brooksdavis**

### CODE REVIEW

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
+=== libsys
+
+Contact: Brooks Davis <brooks@FreeBSD.org>
+
+The libsys project removes direct system calls from `libc.so` and
+`libpthread.so` (aka `libthr.so`) to a seperate `libsys.so`.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:6]: s/seperate/separate/

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+This will:
+
+ * Isolate language runtimes from the details of system call implementations.
+ * Better support logging and replay frameworks for systems calls.
+ * Support elimination of the ability to make system calls outside trusted code in the runtime linker and `libsys`.
+
+This work was initially inspired by a compartmentalization prototype in CheriBSD in 2016.
+Ali Mashtizadeh and Tal Garfinkel picked that work up and attempted to upstream it link:https://reviews.freebsd.org/D14609[D14609].
+Unfortunatly we couldn't figure out how to review and land the massive reorganization required through a phabricator review so it languished.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:15]: s/Unfortunatly/Unfortunately/

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+Since then I've made a number of refinements to the way we link libsys as well as which symbols are provided in which library.
+
+Thanks to Konstantin Belousov <kib@FreeBSD.org> for many round of review and feedback as well as runtime linker fixes.
+Thanks to Mark Johnston <markj@FreeBSD.org> for runtime linker debugging and Dimitry Andric <dim@FreeBSD.org> for sanitizer fixes.  
+Thanks also to everyone how reported bugs and helped debug issues.
+
+==== Known issues (as of the end of the reporting period)
+
+ * The `libsys` ABI is not yet considered stable (it's safe to assume `__sys_foo()` will be supported so language runtimes can use it now).
+ * Programs using address sanitizer must be linked with `-lsys` (resolved in base at publication time.)
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:27]: s/address/the address/
s/time\\.\\)/time)./

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+==== TODO
+
+ * Add a `libsys.h`.  (See link:https://reviews.freebsd.org/D44387[D44387] and other reviews in the stack.)
+ * Update man:intro[2] for `libsys`.
+ * Finalized ABI.  I'm likely to reduce the set of `_` (underscore) prefixed symbols we expose.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:33]: s/Finalized/Finalize the/

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ * MFC the existance of `libsys`?  It's not clear this is practical, but it might be possible to MFC something useful for language runtimes.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:34]: s/existance/existence/

**Review by pauamma — COMMENTED** [review]
Spelling, some wording tweaks. Other than those, it LGTM.

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ * Support elimination of the ability to make system calls outside trusted code in the runtime linker and `libsys`.
+
+This work was initially inspired by a compartmentalization prototype in CheriBSD in 2016.
+Ali Mashtizadeh and Tal Garfinkel picked that work up and attempted to upstream it link:https://reviews.freebsd.org/D14609[D14609].
+Unfortunately we couldn't figure out how to review and land the massive reorganization required through a phabricator review so it languished.
+Last year the CHERI project once again found a need for system call seperation in a new library-based compartmentalization framework in CheriBSD so I rebuilt the patch from scratch, committing dozens of libc cleanups along the way.
+I landed the first batch of changes on February 5th.
+Since then I've made a number of refinements to the way we link libsys as well as which symbols are provided in which library.
+
+Thanks to Konstantin Belousov <kib@FreeBSD.org> for many round of review and feedback as well as runtime linker fixes.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:20]:
```suggestion
Thanks to Konstantin Belousov <kib@FreeBSD.org> for many rounds of review and feedback as well as runtime linker fixes.
```

**Review by pauamma — APPROVED** [review]
With these (missed them the first time), I think it's good to go.

File: website/content/en/status/report-2024-01-2024-03/libsys.adoc [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+This work was initially inspired by a compartmentalization prototype in CheriBSD in 2016.
+Ali Mashtizadeh and Tal Garfinkel picked that work up and attempted to upstream it (link:https://reviews.freebsd.org/D14609[D14609]).
+Unfortunately we couldn't figure out how to review and land the massive reorganization required through a phabricator review so it languished.
+Last year the CHERI project once again found a need for system call separation in a new library-based compartmentalization framework in CheriBSD so I rebuilt the patch from scratch, committing dozens of libc cleanups along the way.
+I landed the first batch of changes on February 5th.
+Since then I've made a number of refinements to the way we link libsys as well as which symbols are provided in which library.
+
+Thanks to Konstantin Belousov <kib@FreeBSD.org> for many rounds of review and feedback as well as runtime linker fixes.
+Thanks to Mark Johnston <markj@FreeBSD.org> for runtime linker debugging and Dimitry Andric <dim@FreeBSD.org> for sanitizer fixes.  
+Thanks also to everyone how reported bugs and helped debug issues.
```
Comment (pauamma) [inline website/content/en/status/report-2024-01-2024-03/libsys.adoc:22]:
```suggestion
Thanks also to everyone who reported bugs and helped debug issues.
```

**Review by pauamma — APPROVED** [review]
I think - I hope :-( - this is the last one.

**Comment by brooksdavis** [comment]
> I think - I hope :-( - this is the last one.

We at least seem to be converging...

**Review by pauamma — APPROVED** [review]

**Comment by lsalvadore** [comment]
Merged with a few minor changes:
- one sentence per line;
- use of `mailto` macro;
- use of `[.filename]` macro;
- avoid contractions;
- formatting consistency.

**Pull request closed without merging**